src/odemis/gui/cont/acquisition.py

```diff
@@ -449,6 +453,433 @@ def open_acquisition_dialog(self):
+                    self._acquiStreams.value.remove(item_stream)
+                # unsubscribe the settings VA's of this removed stream
+                for va in self._get_settings_vas(item_stream):
+                    va.unsubscribe(self._on_settings_va)
+                self._update_acquisition_time()
+                # unsubscribe the name and wavelength VA's of this removed stream
+                for va in self._get_wavelength_vas(item_stream):
+                    va.unsubscribe(self._on_stream_wavelength)
+                item_stream.name.unsubscribe(self._on_stream_name)
+
+        # for every stream in .streams, is it already present in the list?
+        item_stream = [
+            self._panel.streams_chk_list.GetClientData(i)
+            for i in range(self._panel.streams_chk_list.GetCount())
+        ]
+        for s, i in zip(streams, range(len(streams))):
+            if s not in item_stream and not isinstance(s, StaticStream):
+                self._panel.streams_chk_list.Insert(s.name.value, i, s)
+                self._panel.streams_chk_list.Check(i)
+                self._acquiStreams.value.append(s)
+                # subscribe the name and wavelength VA's of this added stream
+                for va in self._get_wavelength_vas(s):
+                    va.subscribe(self._on_stream_wavelength, init=False)
+                s.name.subscribe(self._on_stream_name, init=False)
+                # subscribe the settings VA's of this added stream
+                for va in self._get_settings_vas(s):
+                    va.subscribe(self._on_settings_va)
+                self._update_acquisition_time()
+
+    @wxlimit_invocation(1)  # max 1/s
+    def _update_acquisition_time(self):
+        """
+        Updates the estimated time
+        required for acquisition
+        """
+        acq_time = acqmng.estimateTime(self._acquiStreams.value)
+        acq_time = math.ceil(acq_time)  # round a bit pessimistic
+        # TODO take into account the time of acquiring the zlevels
+        # as well as the time required for moving the actuator
+        txt = u"Estimated time: {}.".format(units.readable_time(acq_time, full=False))
+        self._panel.txt_cryosecom_est_time.SetLabel(txt)
+
+    @call_in_wx_main
+    def _on_stream_wavelength(self, _):
+        """
+        a VA callback that is called when the user changes
+         the emission or excitation wavelength
+        """
+        counts = self._panel.streams_chk_list.GetCount()
+        streams = [
+            self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
+        ]
+        for i, s in zip(range(counts), acqmng.sortStreams(streams)):
+            # replace the unsorted streams with the sorted streams one by one
+            self._panel.streams_chk_list.Delete(i)
+            self._panel.streams_chk_list.Insert(s.name.value, i, s)
+            if s in self._acquiStreams.value:
+                self._panel.streams_chk_list.Check(i)
```

Comment:
It seems the function clears the checklist and repopulates it with the sorted live streams, so maybe it'd be more readable to write it like this:

        self._panel.streams_chk_list.Clear()
        live_streams = [s for s in self._tab_data.streams.value if isinstance(s, LiveStream)]
        for i, stream in enumerate(acqmng.sortStreams(live_streams)):
	        self._panel.streams_chk_list.Insert(stream.name.value, i, stream)
                if stream in self._acquiStreams.value:
		        self._panel.streams_chk_list.Check(i)

Reply:
Using `GetClientData` is interesting because it returns the stream itself, not only its name as a string. In addition, if we use `GetClientData`, we do not need to check if the stream is `LiveStream`